dex/pair/README.md

```diff
@@ -213,3 +213,173 @@ The interaction scripts are located in the _dex/interaction_ directory. The scri
+- __Error Handling:__ To maintain data integrity, a request for a price observation older than the oldest stored observation will result in a SC error. This mechanism helps to prevent the use of outdated or non-existent data.
+- __Versatile Safe Price Request Inputs:__ The mechanism offers several view functions, each providing a different way to request the safe price. These views give users flexibility in querying the safe price by either providing all necessary parameters or using default ones. 
+
+## Endpoints
+
+### getSafePrice
+
+```rust
+    #[view(getSafePrice)]
+    fn get_safe_price(
+        &self,
+        pair_address: ManagedAddress,
+        start_round: Round,
+        end_round: Round,
+        input_payment: EsdtTokenPayment,
+    ) -> EsdtTokenPayment
+```
+
+This function allows you to retrieve the safe price by specifying the pair address, start round, end round, and input payment. 
+It returns the corresponding output payment computed at the safe price.
+
+### getSafePriceByRoundOffset
+
+```rust
+    #[view(getSafePriceByRoundOffset)]
+    fn get_safe_price_by_round_offset(
+        &self,
+        pair_address: ManagedAddress,
+        round_offset: u64,
+        input_payment: EsdtTokenPayment,
+    ) -> EsdtTokenPayment
+```
+
+This function allows you to retrieve the safe price by specifying the pair address, round offset, and input payment. It calls the generic __getSafePrice__ endpoint, by automatically computing the end_round parameter as the current block round, and the start_round as the difference between the current_round and the provided round_offset.
+It returns the corresponding output payment computed at the safe price.
+
+### getSafePriceByDefaultOffset
+
+```rust
+    #[view(getSafePriceByDefaultOffset)]
+    fn get_safe_price_by_default_offset(
+        &self,
+        pair_address: ManagedAddress,
+        input_payment: EsdtTokenPayment,
+    ) -> EsdtTokenPayment
+```
+
+This function allows you to retrieve the safe price by specifying the pair address and input payment. It works in the same way as __getSafePriceByRoundOffset__ endpoint, but instead of using a provided round_offset parameter, it uses a default one, which at this moment is set to 600 rounds (which translates in an one hour window).
+It returns the corresponding output payment computed at the safe price.
+
+### getSafePriceByTimestampOffset
+
+```rust
+    #[view(getSafePriceByTimestampOffset)]
+    fn get_safe_price_by_timestamp_offset(
+        &self,
+        pair_address: ManagedAddress,
+        timestamp_offset: u64,
+        input_payment: EsdtTokenPayment,
+    ) -> EsdtTokenPayment
+```
+
+A more specific view function, it allows you to retrieve the safe price by specifying the pair address, a timestamp offset, and input payment. It basically converts the timestamp_offset value to a generic round_offset, by dividing it with the constant value __SECONDS_PER_ROUND__.
```

Comment:
Something to note for the future & timestamp related implementations: SECONDS_PER_ROUND is not a permanent constant. If protocol changes this (and I guarantee it will at some point), we'll start feeding incorrect averages until we upgrade the view contract.

Reply:
Added a information point:
__Important. The output of this endpoint (and any timestamp related endpoint) will return reliable data as long as the timestamp constant will remain unchanged at the protocol level.__